src/groupeddataframe/groupeddataframe.jl

```diff
@@ -410,3 +399,95 @@ function Base.get(gd::GroupedDataFrame, key::Union{Tuple, NamedTuple}, default)
+    new_starts = gd.starts[idxs]
+    new_ends = gd.ends[idxs]
+    if !allunique(new_starts)
+        throw(ArgumentError("duplicates in idxs argument are not allowed"))
+    end
+    new_groups = zeros(Int, length(gd.groups))
+    idx = gd.idx
+    for i in eachindex(new_starts)
+        @inbounds for j in new_starts[i]:new_ends[i]
+            new_groups[idx[j]] = i
+        end
+    end
+    GroupedDataFrame(gd.parent, gd.cols, new_groups, gd.idx,
+                     new_starts, new_ends, length(new_starts))
+end
+
+# Colon (creates copy)
+Base.getindex(gd::GroupedDataFrame, idxs::Colon) =
+    GroupedDataFrame(gd.parent, gd.cols, gd.groups, gd.idx,
+                     gd.starts, gd.ends, gd.ngroups)
+
+# Convert an array of eltype other than <: Integer to an array of integer indices
+_to_index_array(gd::GroupedDataFrame, idxs::AbstractVector{<:GroupKey}) = [Base.to_index(gd, k) for k in idxs]
+_to_index_array(gd::GroupedDataFrame, idxs::AbstractVector{<:Tuple}) = [Base.to_index(gd, k) for k in idxs]
+_to_index_array(gd::GroupedDataFrame, idxs::AbstractVector{<:NamedTuple}) = [Base.to_index(gd, k) for k in idxs]
+
+# Non-definite eltype, infer and enforce homogeneity
+function _to_index_array(gd::GroupedDataFrame, idxs::AbstractVector)
+    # Edge case - array is empty
+    isempty(idxs) && return Int[]
+
+    # Infer eltype based on type of first index
+    idx1 = idxs[1]
+    T1 = typeof(idx1)
+
+    T = if T1 <: Integer && T1 !== Bool
+        Integer
+    elseif T1 <: GroupKey
+        GroupKey
+    elseif T1 <: Tuple
+        Tuple
+    elseif T1 <: NamedTuple
+        NamedTuple
+    else
+        throw(ArgumentError("Invalid index: $idx1 of type $T1"))
+    end
+
+    ints = zeros(Int, length(idxs))
+    for (i, idx) in enumerate(idxs)
+        idx isa GroupIndexTypes || throw(ArgumentError("Invalid index: $idx1 of type $T1"))
+        idx isa T || throw(ArgumentError("Mixed index types in array not allowed"))
+        ints[i] = Base.to_index(gd, idx)
+    end
+
+    return ints
+end
+
+# Array of non-integer eltype
+Base.getindex(gd::GroupedDataFrame, idxs::AbstractVector) = gd[_to_index_array(gd, idxs)]
+
+
+#
+# Indexing with InvertedIndex
+#
+
+# InvertedIndex wrapping scalar index
+function Base.getindex(gd::GroupedDataFrame, idx::InvertedIndex{<:GroupIndexTypes})
+    selected = trues(length(gd))
+    selected[Base.to_index(gd, idx.skip)] = false
+    return gd[selected]
+end
+
+# InvertedIndex wrapping array
+function Base.getindex(gd::GroupedDataFrame, idx::InvertedIndex{<:AbstractVector})
+    selected = trues(length(gd))
+    selected[_to_index_array(gd, idx.skip)] .= false
+    return gd[selected]
+end
+
+# InvertedIndex wrapping a boolean array
+Base.getindex(gd::GroupedDataFrame, idx::InvertedIndex{<:AbstractVector{Bool}}) = gd[.!idx.skip]
+
+# InvertedIndex wrapping a colon (gives empty GroupedDataFrame)
+Base.getindex(gd::GroupedDataFrame, idx::InvertedIndex{Colon}) = gd[Int[]]
```

Comment:
the fallback method for `Not` seems to be still missing.

Reply:
This is covered by the new method.